src/main/java/net/neoforged/neoforge/common/extensions/IAttributeExtension.java

```diff
@@ -0,0 +1,217 @@
+     */
+    default Component getDebugInfo(AttributeModifier modif, TooltipFlag flag) {
+        Component debugInfo = CommonComponents.EMPTY;
+
+        if (flag.isAdvanced()) {
+            // Advanced Tooltips show the underlying operation and the "true" value. We offset MULTIPLY_TOTAL by 1 due to how the operation is calculated.
+            double advValue = (modif.operation() == Operation.ADD_MULTIPLIED_TOTAL ? 1 : 0) + modif.amount();
+            String valueStr = FORMAT.format(advValue);
+            String txt = switch (modif.operation()) {
+                case ADD_VALUE -> advValue > 0 ? String.format("[+%s]", valueStr) : String.format("[%s]", valueStr);
+                case ADD_MULTIPLIED_BASE -> advValue > 0 ? String.format("[+%sx]", valueStr) : String.format("[%sx]", valueStr);
+                case ADD_MULTIPLIED_TOTAL -> String.format("[x%s]", valueStr);
+            };
+            debugInfo = Component.literal(" ").append(Component.literal(txt).withStyle(ChatFormatting.GRAY));
+        }
+        return debugInfo;
+    }
+
+    /**
+     * Gets the specific UUID that represents a "base" (green) modifier for this attribute.
+     *
+     * @param modif The attribute modifier being checked.
+     * @param flag  The tooltip flag.
+     * @return The UUID of the "base" modifier, or null, if no such modifier may exist.
+     */
+    @Nullable
+    default ResourceLocation getBaseId() {
+        if (this == Attributes.ATTACK_DAMAGE.value()) return AttributeUtil.BASE_ATTACK_DAMAGE_ID;
+        else if (this == Attributes.ATTACK_SPEED.value()) return AttributeUtil.BASE_ATTACK_SPEED_ID;
+        else if (this == Attributes.ENTITY_INTERACTION_RANGE.value()) return AttributeUtil.BASE_ENTITY_REACH_ID;
+        return null;
+    }
+
+    /**
+     * Converts an attribute modifier into its tooltip representation.
+     * <p>
+     * This method does not handle formatting of "base" modifiers, such as Attack Damage or Attack Speed.
+     * <p>
+     *
+     * @param modif The attribute modifier being converted to a component.
+     * @param flag  The tooltip flag.
+     * @return The component representation of the passed attribute modifier.
+     */
+    default MutableComponent toBaseComponent(double value, double entityBase, boolean merged, TooltipFlag flag) {
+        Attribute attr = self();
+
+        Component debugInfo = CommonComponents.EMPTY;
+
+        if (flag.isAdvanced() && !merged) {
+            // Advanced Tooltips cause us to emit the entity's base value and the base value of the item.
+            debugInfo = Component.literal(" ")
+                    .append(Component.translatable("neoforge.adv.base", FORMAT.format(entityBase), FORMAT.format(value - entityBase)).withStyle(ChatFormatting.GRAY));
+        }
+
+        MutableComponent comp = Component.translatable("attribute.modifier.equals.0", FORMAT.format(value), Component.translatable(attr.getDescriptionId()));
+
+        return comp.append(debugInfo);
```

Comment:
Can we just declare `MutableComponent comp` towards the top of this method, and then append in the branch that sets `debugInfo` rather than appending a component that is most likely empty here? I know it doesn't *really* matter given `MutableComponent#append` NO-OPs for empty components, but I think it would lead to cleaner code.